components/authz-service/storage/postgres/migration/sql/58_use_project_db_id_in_references_of_projects.up.sql

```diff
@@ -0,0 +1,159 @@
+BEGIN;
+
+ALTER TABLE iam_statement_projects
+    DROP CONSTRAINT iam_statement_projects_project_id_fkey;
+
+ALTER TABLE iam_statement_projects RENAME COLUMN project_id TO project_temp_id;
+
+ALTER TABLE iam_statement_projects ADD COLUMN project_id SERIAL;
+
+UPDATE
+    iam_statement_projects t
+SET
+    project_id = (
+        SELECT
+            db_id
+        FROM
+            iam_projects
+        WHERE
+            id = t.project_temp_id);
+
+ALTER TABLE iam_statement_projects DROP COLUMN project_temp_id;
+
+ALTER TABLE iam_statement_projects
+    ADD CONSTRAINT "iam_statement_projects_statement_id_project_id_unique" UNIQUE (statement_id, project_id);
+
+CREATE OR REPLACE FUNCTION query_policy (_policy_id TEXT, _projects_filter TEXT[])
+    RETURNS json
+    AS $$
+    WITH temp AS (
+        SELECT
+            pol.db_id,
+            pol.id,
+            pol.name,
+            pol.type,
+            -- get policy's statements using temporary table
+            ( WITH statement_rows AS (
+                    SELECT
+                        stmt.id,
+                        stmt.effect,
+                        stmt.actions,
+                        stmt.resources,
+                        stmt.role,
+                        -- get each statement's projects by cross-referencing iam_policy_statements and iam_statement_projects
```

Comment:
we did this in other migrations, too, but this comment is invalid bc we nixed `iam_policy_statements`